esphome/components/debug/debug_component.cpp

```diff
@@ -195,16 +225,70 @@ void DebugComponent::dump_config() {
   ESP_LOGD(TAG, "Flash Chip ID=0x%08X", ESP.getFlashChipId());
   ESP_LOGD(TAG, "Reset Reason: %s", ESP.getResetReason().c_str());
   ESP_LOGD(TAG, "Reset Info: %s", ESP.getResetInfo().c_str());
+
+  device_info += "|Chip: 0x" + uint32_to_string(ESP.getChipId());
+  device_info += "|SDK: ";
+  device_info += ESP.getSdkVersion();
+  device_info += "|Core: ";
+  device_info += ESP.getCoreVersion().c_str();
+  device_info += "|Boot: ";
+  device_info += to_string(ESP.getBootVersion());
+  device_info += "|Mode: " + to_string(ESP.getBootMode());
+  device_info += "|CPU: " + to_string(ESP.getCpuFreqMHz());
+  device_info += "|Flash: 0x" + uint32_to_string(ESP.getFlashChipId());
+  device_info += "|Reset: ";
+  device_info += ESP.getResetReason().c_str();
+  device_info += "|";
+  device_info += ESP.getResetInfo().c_str();
 #endif
+
+  if (this->device_info_ != nullptr) {
+    if (device_info.length() > 255)
+      device_info.resize(255);
+    this->device_info_->publish_state(device_info);
+  }
 }
+
 void DebugComponent::loop() {
+  // calculate loop time - from last call to this one
+  if (this->loop_time_sensor_ != nullptr) {
+    uint32_t now = millis();
+    uint32_t loop_time = now - this->last_loop_timetag_;
+    this->max_loop_time_ = max(this->max_loop_time_, loop_time);
+    this->last_loop_timetag_ = now;
+  }
+}
+
+void DebugComponent::update() {
   uint32_t new_free_heap = ESP.getFreeHeap();
   if (new_free_heap < this->free_heap_ / 2) {
     this->free_heap_ = new_free_heap;
     ESP_LOGD(TAG, "Free Heap Size: %u bytes", this->free_heap_);
     this->status_momentary_warning("heap", 1000);
   }
+
+  if (this->free_sensor_ != nullptr) {
+    this->free_sensor_->publish_state(new_free_heap);
+  }
+
+// CLANG_TIDY uses an old arduino framework which doesn't support the heap state functions
+#if defined(ARDUINO_ARCH_ESP8266) & !defined(CLANG_TIDY)
+  // NOTE: Requires arduino_version 2.5.2 or above
```

Comment:
With the newly-updated dev branch you merged in, we can actually check the Arduino version now:
```suggestion
#if defined(ARDUINO_ARCH_ESP8266) && ARDUINO_VERSION_CODE >= VERSION_CODE(2, 5, 2)
```